101-data-lake-store-encryption-key-vault/azuredeploy.json

```diff
@@ -1,65 +1,127 @@
+      "metadata": {
+        "description": "The location in which to create the Data Lake Store account."
+      }
+    },
+    "keyVaultName": {
+      "type": "string",
+      "metadata": {
+        "description": "The Azure Key Vault name."
+      }
+    },
+    "keyVaultResourceGroupName": {
+      "type": "string",
+      "metadata": {
+        "description": "The Azure Key Vault resource group name."
+      }
+    },
+    "keyName": {
+      "type": "string",
+      "metadata": {
+        "description": "The Azure Key Vault encryption key name."
+      }
+    },
+    "keyVersion": {
+      "type": "string",
+      "metadata": {
+        "description": "The Azure Key Vault encryption key version."
+      }
+    }
+  },
+  "resources": [
+    {
+      "apiVersion": "2016-11-01",
+      "name": "[parameters('dataLakeStoreName')]",
+      "location": "[parameters('location')]",
+      "type": "Microsoft.DataLakeStore/accounts",
+      "properties": {
+        "encryptionState": "Enabled",
+        "encryptionConfig": {
+          "type": "UserManaged",
+          "keyVaultMetaInfo": {
+            "keyVaultResourceId": "[resourceId(parameters('keyVaultResourceGroupName'), 'Microsoft.KeyVault/vaults', parameters('keyVaultName'))]",
+            "encryptionKeyName": "[parameters('keyName')]",
+            "encryptionKeyVersion": "[parameters('keyVersion')]"
+          }
         }
+      },
+      "identity": {
+        "type": "SystemAssigned"
+      }
+    },
+    {
+      "type": "Microsoft.KeyVault/vaults/accessPolicies",
+      "name": "[concat(parameters('keyVaultName'), '/add')]",
+      "apiVersion": "2015-06-01",
+      "dependsOn": [
+        "[resourceId('Microsoft.DataLakeStore/accounts', parameters('dataLakeStoreName'))]"
+      ],
+      "properties": {
+        "accessPolicies": [
+          {
+            "objectId": "[reference(resourceId('Microsoft.DataLakeStore/accounts', parameters('dataLakeStoreName')), '2016-11-01', 'Full').identity.principalId]",
+            "tenantId": "[reference(resourceId('Microsoft.DataLakeStore/accounts', parameters('dataLakeStoreName')), '2016-11-01', 'Full').identity.tenantId]",
+            "permissions": {
+              "keys": [
+                "encrypt",
+                "decrypt"
+              ]
+            }
+          }
+        ]
+      }
     },
-    "resources": [
-        {
-            "apiVersion": "2016-11-01",
-            "name": "[parameters('accountName')]",
-            "location": "[parameters('location')]",
-            "type": "Microsoft.DataLakeStore/accounts",
-            "properties": {
-                "newTier": "Consumption",
-                "encryptionState": "Enabled",
+    {
+      "type": "Microsoft.Resources/deployments",
+      "apiVersion": "2017-05-10",
+      "name": "updateAdlsAccount",
+      "dependsOn": [
+        "[resourceId(parameters('keyVaultResourceGroupName'), 'Microsoft.KeyVault/vaults/accessPolicies', parameters('keyVaultName'), 'add')]"
```

Comment:
I think your validation error is coming from here, try:

[concat(parameters('keyVaultName'), '/add')]